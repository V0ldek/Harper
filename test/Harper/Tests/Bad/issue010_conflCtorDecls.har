value List a = {
  variant Empty = { };
  
  variant NonEmpty = {
    data = {
      head :: a;
    }
  };

  variant NonEmpty = {
    data = {
      head :: a;
      tail :: List a;
    }
  };
};

value Bag a b c d = {
  variant Empty = { };
  variant Bag = {
    data = {
      a :: a; a :: b; c :: c; c :: d;
    }
  };
};


main :: Integer;
main = 42;