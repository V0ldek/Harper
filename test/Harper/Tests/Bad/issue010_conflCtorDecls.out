Error: conflicting ctor names `Empty`, `NonEmpty`.
Conflicting declarations:
value Bag a b c d = {
  variant Empty = {
    } ;
  variant Bag = {
    data = {
      a :: a ;
      a :: b ;
      c :: c ;
      c :: d ;
      }
    } ;
  }

Located at line 18 column 1
value List a = {
  variant Empty = {
    } ;
  variant NonEmpty = {
    data = {
      head :: a ;
      }
    } ;
  variant NonEmpty = {
    data = {
      head :: a ;
      tail :: List a ;
      }
    } ;
  }

Located at line 1 column 1
value List a = {
  variant Empty = {
    } ;
  variant NonEmpty = {
    data = {
      head :: a ;
      }
    } ;
  variant NonEmpty = {
    data = {
      head :: a ;
      tail :: List a ;
      }
    } ;
  }

Located at line 1 column 1
value List a = {
  variant Empty = {
    } ;
  variant NonEmpty = {
    data = {
      head :: a ;
      }
    } ;
  variant NonEmpty = {
    data = {
      head :: a ;
      tail :: List a ;
      }
    } ;
  }

Located at line 1 column 1


Execution terminated with an error: type error.